front/src/modules/trainschedule/components/TimetableV2/hooks.ts

```diff
@@ -151,7 +151,7 @@ const useTrainSchedulesDetails = (
             id: trainSchedule.id,
             trainName: trainSchedule.train_name,
             startTime: dayjs(trainSchedule.start_time).format('D/MM/YYYY HH:mm:ss'), // format to time
-            stepsCount: trainSchedule.schedule?.length ?? 0,
+            stopsCount: (trainSchedule.schedule?.length ?? 0) + 1, // +1 to take the final stop (destination) into account
```

Comment:
Here we check only if there are trains in trainSchedule.schedule but what about if one of them has (the first schedule for example) trainSchedule.schedule[0].stop_for is null/undefined ? It will still count as a stopCount but we don't want to.
We should probably create a const to count the number of schedules with a stop_for > 0.